front/src/common/Map/WarpedMap/SimulationWarpedMap.tsx

```diff
@@ -0,0 +1,322 @@
+      // 0 width, and we just need to specify the middle longitude (based on the visible part of the path on the screen,
+      // so depending on the latTop and latBottom values):
+      const clippedPath = clip(transformedPath, {
+        type: 'rectangle',
+        points: [
+          [regularBBox[0], latTop],
+          [regularBBox[2], latBottom],
+        ],
+      }) as typeof transformedPath;
+      const clippedPathBBox = bbox(clippedPath) as BBox2d;
+      const lngAverage = (clippedPathBBox[0] + clippedPathBBox[2]) / 2;
+
+      return [
+        [lngAverage, latTop],
+        [lngAverage, latBottom],
+      ] as LngLatBoundsLike;
+    }
+
+    if (state.type === 'dataLoaded' || state.type === 'pathLoaded') {
+      const { regularBBox } = state;
+      const lngAverage = (regularBBox[0] + regularBBox[2]) / 2;
+
+      return [
+        [lngAverage, regularBBox[1]],
+        [lngAverage, regularBBox[3]],
+      ];
+    }
+
+    // This should never occur:
+    return [
+      [0, 0],
+      [0, 0],
+    ] as LngLatBoundsLike;
+  }, [chart, state]);
+
+  // Itinerary handling:
+  const {
+    positionValues,
+    timePosition,
+    allowancesSettings,
+    simulation: { present: simulation },
+  } = useSelector((s: RootState) => s.osrdsimulation);
+  const selectedTrain = useSelector(getSelectedTrain);
+  const itineraryState: AsyncMemoState<Feature<LineString> | null> = useAsyncMemo(async () => {
+    if (!selectedTrain) return null;
+    if (state.type !== 'dataLoaded') return null;
+
+    const foundTrain = simulation.trains.find((train) => train.id === selectedTrain.id);
+    if (!foundTrain) return null;
+
+    const { data: path } = await getPath({ id: foundTrain.path });
+    if (!path) return null;
+
+    return lineString(path.geographic.coordinates);
+  }, [selectedTrain, state.type, simulation]);
+  const warpedItinerary = useMemo(() => {
+    const itinerary = getAsyncMemoData(itineraryState);
+    if (itinerary && state.type === 'dataLoaded') return state.transform(itinerary) || undefined;
+    return undefined;
+  }, [itineraryState, state]);
+
+  // Trains handling:
+  const trainsIndex = useMemo(() => keyBy(simulation.trains, 'id'), [simulation.trains]);
+  const trainsPositionsState: AsyncMemoState<
+    (TrainPosition & { train: Train; isSelected?: boolean })[]
+  > = useAsyncMemo(async () => {
+    const path = getAsyncMemoData(itineraryState);
+    if (!path || state.type !== 'dataLoaded') return [];
+
+    const transformedPath = state.transform(path) as typeof path;
+    return getSimulationHoverPositions(
+      path,
+      simulation,
+      timePosition,
+      positionValues,
+      selectedTrain?.id,
+      allowancesSettings
+    ).map((position) => {
+      const transformedTrain = { ...position };
+      const pathLength = length(path);
+      const transformedPathLength = length(transformedPath);
```

Comment:
These computations can be done outside the map, just before it
```suggestion
      const pathLength = length(path);
      const transformedPathLength = length(transformedPath);
```

Reply:
Good catch